File: data_structures/python/Queue.py
import Stack as S

class Queue:
    def __init__(self, size):
        self.s1 = S.Stack(size)
        self.s2 = S.Stack(size)

    def enqueue(self, element):
        self.s1.push(element)

    def dequeue(self):
        if self.getSize() == 0:
            print('queue is empty')
            return

        if self.s2.isEmpty():
            while not self.s1.isEmpty():
                self.s2.push(self.s1.pop())
        return self.s2.pop()

    def peek(self):
        if self.getSize() == 0:
            print('queue is empty')
            return

        return self.s2.top()

    def getSize(self):
        return self.s1.getSize() + self.s2.getSize()

    def traverse(self):
        self.s2.traverse()

my_queue = Queue(5)
my_queue.enqueue(1)
my_queue.enqueue(2)
my_queue.enqueue(3)
my_queue.enqueue(4)
my_queue.traverse()

File: data_structures/java/Queue.java
public class Queue {
    private Stack s1;
    private Stack s2;
    int front, rear;

    public Queue(int size) {
        this.s1 = new Stack(size);
        this.s2 = new Stack(size);
    }

    public void enqueue(Object element) {
        s1.push(element);
    }

    public Object dequeue() {
        if (getSize() == 0) {
            return null;
        }
        if (s2.isEmpty()) {
            while(!s1.isEmpty()) {
                s2.push(s1.pop());
            }
        }
        return s2.pop();
    }

    public int getSize() {

        return s1.getSize() + s2.getSize();
    }

    public Object peek() {
        if(getSize() == 0) {
            return null;
        }
        return s2.top();
    }
    public void traverse() {
        if (getSize() == 0) {
            System.out.println("Queue is empty");
            return;
        }
        s2.traverse();
    }
    public static void main(String[] args) {
        Queue q = new Queue(3);
        q.enqueue(1);
        q.enqueue(2);
        q.enqueue(3);
        q.dequeue();
        q.peek();
        q.traverse();
    }
}
